crates/html-macro-test/src/lib.rs

```diff
@@ -39,13 +32,13 @@ fn empty_div() {
 fn one_prop() {
     let mut props = HashMap::new();
     props.insert("id".to_string(), "hello-world".to_string());
-    let mut expected = VirtualNode::new("div");
+    let mut expected = VirtualNode::element_variant("div");
```

Comment:
Feeling as I'm reading this is that `element` vs `element_variant` feels confusing and maybe a tad too much.

I'd almost rather something like

```rust
let node = VirtualNode::element("div");
node.as_elem_mut().unwrap().props = props;
```

In order to ditch the `*_variant` methods entirely.

Thoughts? Maybe you disagree?

Reply:
Generally I would have ditched the `element_variant` and `text_variant` methods in favor of the `VirtualNodeElement` and `VirtualNodeText` types. But the prelude only exports `VirtualNode` and I'm trying to avoid polluting the prelude too much. On the other hand that's only relevant within the html macro, outside of the html macro people can simply import the variant structs directly.

What about using the mutable reference "hack" in the macro implementation and `VirtualNodeElement` and `VirtualNodeText` constructors everywhere else? Then we could get rid of the variant construction functions.

Reply:
Let's continue this discussion here: https://github.com/chinedufn/percy/pull/71#discussion_r253949675

Reply:
I'll trust your judgement here - I think we just need to continue to use the API in real applications and see how it feels to be able to make the "best" decision here.

So sounds good!

Reply:
FWIW I'm also not that worried about adding them to the prelude if it minimizes confusion